org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/references/search/SearchNode.java

```diff
@@ -0,0 +1,147 @@
+/*******************************************************************************
+* Copyright (c) 2023 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package org.eclipse.lemminx.extensions.references.search;
+
+import org.eclipse.lemminx.dom.DOMDocument;
+import org.eclipse.lemminx.dom.DOMNode;
+import org.eclipse.lemminx.dom.DOMRange;
+import org.eclipse.lemminx.extensions.references.search.SearchQuery.Direction;
+
+/**
+ * Search node result.

```

Comment:
If you see code from master, the search collected DOM attribute, text. It worked perfectly because multiple was not implemented : the value to search was the full attribute value or the full content of text node.

For multiple value comes from split of attribute value / text content. The SearchNode represents that. Instead of working with DOMNode, we work with SearchNode. If multiple is false,SearchNode has the same start / end offset of the DOM attribute / text

Reply:
Okay. I think that it's worth adding something like this to the javadoc:

`eg. text node, attribute value node, or in the case of <code>multiple</code>, a node that represents the subsection of the text or attribute node that contains the reference`

Reply:
I add a Javadoc which explains what is a search node. If the explanation is not enough, please tell me.

Reply:
Yep, I think it's a great explanation